app/controllers/followup_controller.rb

```diff
@@ -0,0 +1,189 @@
+# -*- encoding : utf-8 -*-
+class FollowupController < ApplicationController
+  before_filter :check_read_only, :set_incoming_message_and_last_request,
+                  :check_for_external_request, :check_user_credentials,
+                  :check_address, :check_request_matches_incoming_message,
+                  :set_params
+
+  before_filter :check_reedit, :only => [:preview, :create]
+
+  before_filter :check_responses_allowed, :only => [:create]
+
+  def new
+  end
+
+  def create
+    @outgoing_message.info_request = @info_request
+    if @info_request.find_existing_outgoing_message(params[:outgoing_message][:body])
+      flash.clear # otherwise the message sent notice may be redisplayed
+      flash[:error] = _('You previously submitted that exact follow up message for this request.')
+    elsif @outgoing_message.valid?
+      send_followup
+      redirect_to request_url(@info_request) and return
+    end
+    render :action => 'new'
+  end
+
+  def preview
+    @outgoing_message.info_request = @info_request
+    if @outgoing_message.what_doing == 'internal_review'
+       @internal_review = true
+    end
+    unless @outgoing_message.valid?
+      render :action => 'new'
+      return
+    end
+  end
+
+  private
+
+  def check_address
+    if !OutgoingMailer.is_followupable?(@info_request, @incoming_message)
+      raise "unexpected followupable inconsistency" if @info_request.public_body.is_requestable?
+      @reason = @info_request.public_body.not_requestable_reason
+      render :action => 'followup_bad'
+      return
+    end
+  end
+
+  def check_for_external_request
+    if @info_request.is_external?
+      @reason = 'external'
+      render :action => 'followup_bad'
+      return
+    end
+  end
+
+  def check_internal_review
+    @internal_review = false
+    @internal_review_pass_on = false
+    if params[:internal_review]
+      @internal_review = true
+      @internal_review_pass_on = true
+    end
+  end
+
+  def check_reedit
+    if params[:reedit]
+      render :action => 'new'
+      return
+    end
+  end
+
+  def check_request_matches_incoming_message
+    if @incoming_message and @info_request != @incoming_message.info_request
+      raise ActiveRecord::RecordNotFound.
+              new("Incoming message #{@incoming_message.id} does not belong " \
```

Comment:
I... I'm not sure. I take your point but I like that new lines up under the calling object. Will wait for a human adjudicator